### PR TITLE
service: add max records limit to read_all and read_many

### LINKED ISSUE #252: read_all is limited to 10 results
Read all performs a ES search, which by default is limited to 10 results... therefore is not actually reading all but just 10 of them

### PR DESCRIPTION
**Pull request by ppanero**
close #252

### CODE REVIEW

File: invenio_records_resources/services/records/service.py [hunk 1]
```diff
@@ -318,30 +318,36 @@ def _read_many(self, identity, es_query, fields=None, record_cls=None,
...
             search = search.source(fields)
 
+        search = search[0:max_records]
         search_result = search.query(es_query).execute()
 
         return search_result
 
-    def read_many(self, identity, ids, fields=None, **kwargs):
+    def read_many(
+        self, identity, ids, fields=None, max_records=100, **kwargs
```
Comment (ppanero) [inline invenio_records_resources/services/records/service.py:327]: assert len(ids) < max_records

File: invenio_records_resources/services/records/service.py [hunk 2]
```diff
@@ -318,30 +318,36 @@ def _read_many(self, identity, es_query, fields=None, record_cls=None,
...
 
-        results = self._read_many(identity, query, fields, **kwargs)
+        results = self._read_many(
+            identity, query, fields, max_records, **kwargs)
 
         return self.result_list(self, identity, results)
 
-    def read_all(self, identity, fields, cache=True, **kwargs):
+    def read_all(
+        self, identity, fields, cache=True, max_records=100, **kwargs
```
Comment (ppanero) [inline invenio_records_resources/services/records/service.py:341]: keep max_records

**Comment by ppanero** [comment]
unearthed https://github.com/inveniosoftware/invenio-records-resources/issues/257

File: invenio_records_resources/services/records/service.py [hunk 3]
```diff
@@ -291,9 +291,9 @@ def read(self, id_, identity):
             links_tpl=self.links_item_tpl,
         )
 
-    def _read_many(self, identity, es_query, fields=None, record_cls=None,
-                   search_opts=None, extra_filter=None, preference=None,
-                   **kwargs):
+    def _read_many(self, identity, es_query, fields=None, max_records=100,
```
Comment (fenekku) [inline invenio_records_resources/services/records/service.py:294]: :100: was just going to suggest 100 too

**Review by fenekku — APPROVED** [review]

**Pull request merged**